### PR TITLE
feat: snapshot creator in Rust

### PR DESCRIPTION
**Pull request by bartlomieju**
This is first pass at rewriting snapshot creator in Rust.

For now I only got it building using `cargo build`.

CC @ry @piscisaureus

### CODE REVIEW

File: core/snapshot_creator.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    unsafe { std::slice::from_raw_parts(snapshot.data_ptr, snapshot.data_len) };
+
+  let write_result = out_file.write_all(snapshot_slice);
+
+  unsafe {
+    // TODO: calling this method causes
+    //  `error for object 0x7ffee91ccfc0: pointer being freed was not allocated`
+    //  I suspect it's because snapshot is created by Rust and we try to free it in C++
+    //  using drop(snapshot) works fine though
+    // libdeno::deno_snapshot_delete(&mut snapshot);
```
Comment (bartlomieju) [inline core/snapshot_creator.rs]: @ry you might be interested in this bit

Comment (bartlomieju) [reply core/snapshot_creator.rs]: Interestingly build doesn't pass on `asan` reporting memory leak:
```
==7124==ERROR: LeakSanitizer: detected memory leaks
Direct leak of 532072 byte(s) in 1 object(s) allocated from:
    #0 0x558525d3927d in operator new[](unsigned long) /b/swarming/w/ir/k/src/third_party/llvm/compiler-rt/lib/asan/asan_new_delete.cc:108:3
    #1 0x5585273bec2e in v8::internal::Snapshot::CreateSnapshotBlob(v8::internal::SnapshotData const*, v8::internal::SnapshotData const*, std::__1::vector<v8::internal::SnapshotData*, std::__1::allocator<v8::internal::SnapshotData*> > const&, bool) (/home/travis/build/denoland/deno/target/debug/rust_snapshot_creator+0x23b1c2e)
    #2 0x558525dade20 in v8::SnapshotCreator::CreateBlob(v8::SnapshotCreator::FunctionCodeHandling) (/home/travis/build/denoland/deno/target/debug/rust_snapshot_creator+0xda0e20)
    #3 0x558528dfeb40 in deno_snapshot_new core/libdeno/api.cc:119:37
    #4 0x558528dfc593 in rust_snapshot_creator_bin::main::h4268fe93d28c8af3 core/snapshot_creator.rs:75:26
SUMMARY: AddressSanitizer: 532072 byte(s) leaked in 1 allocation(s).
```

**Comment by bartlomieju** [comment]
I'll try to rewrite it once more using `core/isolate.rs` API

**Comment by ry** [comment]
thanks @bartlomieju - I'm going to work on this further in a new PR https://github.com/denoland/deno/pull/2617

**Pull request closed without merging**